### PR TITLE
Update blast 2.9.0 adding rpsbproc among blast tools - replace #21497

### PR DESCRIPTION
**Pull request by Juke34**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/blast/build.sh [hunk 1]
```diff
@@ -15,13 +16,24 @@ if test x"`uname`" = x"Linux"; then
 fi
 
 if [ `uname` == Darwin ]; then
-    export LDFLAGS="${LDFLAGS} -Wl,-rpath,$PREFIX/lib -lz -lbz2"
+    #export LDFLAGS="${LDFLAGS} -Wl,-rpath,$PREFIX/lib -lz -lbz2"
+    export LDFLAGS="${LDFLAGS} -Wl,-rpath,$PREFIX/lib -lz -lbz2 -flat_namespace -headerpad_max_install_names"
+    export CXXFLAGS="$CXXFLAGS -fpascal-strings -fpermissive -Wno-deprecated-register -fno-common"
+    export CFLAGS="$CFLAGS -fpascal-strings -Wno-deprecated-register -fno-common"
```
Comment (mbargull) [inline recipes/blast/build.sh:22]: Can you explain (ideally with comments in the build script 😉) why those additional flags are added?

File: recipes/blast/meta.yaml [hunk 2]
```diff
@@ -6,11 +6,20 @@ package:
...
+  # Reverse Position-Specific BLAST (RPS-BLAST), also known as CD-Search (Conserved
+  # Domain Search).
+  - url: https://ftp.ncbi.nih.gov/pub/mmdb/cdd/rpsbproc/RpsbProc-src.tar.gz
+    sha256: 8b29f7e7d0a0654d10943e362557ed2d8df9a16218a4c8441c4dea4e7dc9154f
+    folder: RpsbProc
 
 build:
-  number: 4
+  number: 5
+  skip: True # [linux]
```
Comment (mbargull) [inline recipes/blast/meta.yaml:22]: I support this is just for testing this PR?

File: recipes/blast/build.sh (hunk 1 above)
Comment (Juke34) [reply recipes/blast/build.sh:22]: Will be difficult, it was a while ago. I have tried hundred of combination to minimise warnings and error when compiling os OSX.
Those changes are maybe not needed anymore with the new SDK

File: recipes/blast/meta.yaml (hunk 2 above)
Comment (Juke34) [reply recipes/blast/meta.yaml:22]: yes it was just to test on OSX as it takes hours to pass the linux test

File: recipes/blast/build.sh (hunk 1 above)
Comment (mbargull) [reply recipes/blast/build.sh:22]: Okay, then I'll try a build without them.

Comment (Juke34) [reply recipes/blast/build.sh]: Yes it works now. Thx

**Comment by Juke34** [comment]
@mbargull it sounds good now, we can merge

**Comment by Juke34** [comment]
@BiocondaBot please add label

**Pull request merged**